Framework/Algorithms/test/ScaleXTest.h

```diff
@@ -314,21 +284,132 @@ class ScaleXTest : public CxxTest::TestSuite {
     return scale.getProperty("OutputWorkspace");
   }
 
+  void testTimeOfFlightAtHistogramIndex(
+      size_t i, Mantid::DataObjects::EventWorkspace_sptr inputWS,
+      boost::shared_ptr<Mantid::API::IEventWorkspace> resultEventWS,
+      double factor) {
+    auto &inEvents = inputWS->getSpectrum(i);
+    auto &outEvents = resultEventWS->getSpectrum(i);
+    TS_ASSERT_EQUALS(outEvents.getNumberEvents(), inEvents.getNumberEvents());
+
+    auto inTOFs = inEvents.getTofs();
+    auto outTOFs = outEvents.getTofs();
+    TS_ASSERT_EQUALS(inTOFs.size(), outTOFs.size());
+
+    for (size_t j = 0; j < inTOFs.size(); ++j) {
+      TS_ASSERT_DELTA(outTOFs[j], factor * inTOFs[j], 1e-12);
+    }
+  }
+
   void testScaleFactorApplied(
       const Mantid::API::MatrixWorkspace_const_sptr &inputWS,
       const Mantid::API::MatrixWorkspace_const_sptr &outputWS, double factor,
       bool multiply) {
-    const size_t xsize = outputWS->blocksize();
+
     for (size_t i = 0; i < outputWS->getNumberHistograms(); ++i) {
+      testScaleFactorAppliedAtHistIndex(i, inputWS, outputWS, factor, multiply);
+    }
+  }
+
+  ///
+  /**
+  Only loops through the bins of the index of the parameter histogram
+  The idea is to be called from within a loop that goes through the histograms
+  like on line 113
+
+  @param i :: Specifies the histogram index
+  */
+  void testScaleFactorAppliedAtHistIndex(
```

Comment:
Can you pass the input and output histograms as arguments instead of the workspaces? e.g

```
void testScaleFactorAppliedAtHistIndex(ize_t i, const Histogram &inputWS,
       const Histogram &outputWS, double factor,
       bool multiply)
```

Then at the top you can have `using Mantid::HistogramData::Histogram;`